Fix parsing of string objects

NobodysNightmare opened the pull request:
## OpenProject work package

Fixes another error from https://community.openproject.org/work_packages/18571

NobodysNightmare committed:
fix parsing of string objects

- accept them since format has changed
- actually revert the URL escaping that has to be performed

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://nobodysnightmare-feature-string-object-parsing.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://nobodysnightmare-feature-string-object-parsing.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

NobodysNightmare commented:
@floriank right right... fixed that

NobodysNightmare committed:
improve style

- fewer lines is always better
- fewer locals is also always better
-> now judge this commit!

[ci skip]
